### PR TITLE
Update README

### LINKED ISSUE #1072: Document in the README how to download restate binaries
Add a one-liner to the README to unzip the cli and server, and add it to the path.

### PR DESCRIPTION
**Pull request by tillrohrmann**
This commit updates various aspects of our README:

* Add install instructions
* Add community section
* Add contributing section
* Update intro part with image and examples
* Link to quickstart and tour of Restate

This fixes #1072.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,48 +33,66 @@ Restate supports the following SDKs:
...
+* **suspending user code**: long-running user code suspends when awaiting on a promise and resume when that promise is resolved.
+* **reliable communication**: user code communicates with exactly-once semantics. Restate reliably delivers messages and anchors both sender and receiver in the durable execution to ensure no losses or duplicates can happen.
+* **durable timers**: user code can sleep (and suspend) or schedule calls for later.
+* **isolation**: user code can be keyed, which makes Restate scheduled them to obey single-writer-per-key semantics.
+* **consistent state**: keyed user code can attach key/value state, which is eagerly pushed into handlers during invocation, and written back upon completion. This is particularly efficient for FaaS deployments (stateful serverless, yay!).
+* **observability & introspection**: Restate automatically generates Open Telemetry traces for the interactions between handlers and gives you a SQL shell to query the distributed state of the application.
```
Comment (AhmedSoliman) [inline README.md:85]:
```suggestion
* **Reliable Execution**: user code will always run to completion. Intermediate failures result in re-tries that use the durable execution mechanism to recover partial progress and not duplicate already executed steps.
* **Suspending User Code**: long-running user code suspends when awaiting on a promise and resume when that promise is resolved.
* **Reliable Communication**: user code communicates with exactly-once semantics. Restate reliably delivers messages and anchors both sender and receiver in the durable execution to ensure no losses or duplicates can happen.
* **Durable Timers**: user code can sleep (and suspend) or schedule calls for later.
* **Isolation**: user code can be keyed, which makes Restate scheduled them to obey single-writer-per-key semantics.
* **Consistent State**: keyed user code can attach key/value state, which is eagerly pushed into handlers during invocation, and written back upon completion. This is particularly efficient for FaaS deployments (stateful serverless, yay!).
* **Observability & Introspection**: Restate automatically generates Open Telemetry traces for the interactions between handlers and gives you a SQL shell to query the distributed state of the application.
```

Comment (tillrohrmann) [reply README.md]: Good improvement. Will update the PR.

File: README.md [hunk 2]
```diff
@@ -31,48 +33,66 @@ Restate supports the following SDKs:
...
 * [Java and Kotlin](https://github.com/restatedev/sdk-java)
 
-## Building Restate
+## Install
 
-In order to build Restate locally [follow the build instructions](https://github.com/restatedev/restate/blob/main/docs/dev/local-development.md#building-restate).
+We offer pre-built binaries of the CLI and the server for MacOS and Linux.
 
-## Running Restate
+### Install the server
```
Comment (slinkydeveloper) [inline README.md:40]: We're missing the docker image, i feel this is something we should have somewhere in the readme, perhaps somewhere here in this paragraph, like:

Pull the container image:

```
docker pull docker.io/restatedev/restate
```

**Review by slinkydeveloper — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (tillrohrmann) [reply README.md:40]: I've pushed an fixup adding how to run the Restate server via `docker`.

**Review by slinkydeveloper — APPROVED** [review]

**Comment by tillrohrmann** [comment]
Thanks for the review @AhmedSoliman and @slinkydeveloper. Merging this PR now.

**Pull request merged**